patroni/postgresql/config.py

```diff
@@ -1154,16 +1156,27 @@ def reload_config(self, config: Dict[str, Any], sighup: bool = False) -> None:
             if self._postgresql.major_version >= 90500:
                 time.sleep(1)
                 try:
-                    pending_restart = self._postgresql.query(
-                        'SELECT COUNT(*) FROM pg_catalog.pg_settings'
-                        ' WHERE pg_catalog.lower(name) != ALL(%s) AND pending_restart',
-                        [n.lower() for n in self._RECOVERY_PARAMETERS])[0][0] > 0
-                    self._postgresql.set_pending_restart(pending_restart)
+                    settings_diff = {
+                        p: (v, (lambda x: '?' if x is None else x)(self._postgresql.get_guc_value(p)))
```

Comment:
What is `(lambda x: '?' if x is None else x)`?

Reply:
`get_guc_value()` returns `None` if it failed to execute the cmd:)